packages/template/src/lib/stack-app/customers/index.ts

```diff
@@ -36,7 +36,11 @@ export type Customer<IsServer extends boolean = false> =
   & {
     readonly id: string,
 
-    createCheckoutUrl(offerIdOrInline: string | InlineOffer): Promise<string>,
+    createCheckoutUrl(
+      ...args: IsServer extends true
+        ? [offerIdOrInlineOffer: string | InlineOffer]
+        : [offerId: string]
+    ): Promise<string>,
```

Comment:
maybe a little late but can we still change this to

```suggestion
    createCheckoutUrl(options: (
      | { offerId: string }
      | (IsServer extends true ? { offer: InlineOffer } : never)
    )): Promise<string>,
```

my thinking being that we might have other types in the future, eg. to create checkout urls for entire groups or multiple offers at the same time